docs-app/ember-cli-build.js

```diff
@@ -28,15 +28,21 @@ module.exports = function (defaults) {
           localIdentName: isProduction()
             ? '[sha512:hash:base64:5]'
             : '[path][name]__[local]',
-          // global mode, can be either `global` or `local`
-          mode: (resourcePath) => {
-            if (resourcePath.includes(`/${app.name}/`)) {
-              return 'local';
-            }
 
-            return 'global';
+          // only enable local mode for the host app itself
+          mode(resourcePath) {
+            // The host app and active child addons are moved into a common
+            // stable temp dir (`options.workspaceDir`), before the `css-loader`
+            // processes them.
+            // We only want to enable local mode for our own host app. All other
+            // addons should be loaded in global mode.
+            const hostAppWorkspaceDir = `${options.workspaceDir}/${app.name}`;
+            const isHostAppPath = resourcePath.startsWith(hostAppWorkspaceDir);
```

Comment:
I think `startsWith()` doesn't currently work when the host app sets the environment variable `SAVE_WORKSPACE`, either to [temporarily get around the `workspaceDir` bug](https://github.com/embroider-build/embroider/issues/1297), or for possibly another reason.

```ts
const workspaceDir = 'my-workspace';
process.env.SAVE_WORKSPACE = workspaceDir;

module.exports = function (defaults) {
  const app = new EmberApp(...);

  const options = {
    packageOptions: { ... },
    ...
    workspaceDir,
};
```

```sh
# Sample output
[docs-app] /Users/isaaclee/Documents/GitHub/ember-container-query/docs-app/my-workspace/docs-app/components/widgets/widget-4/memo.css
[docs-app] my-workspace/docs-app
[docs-app] global
[docs-app] /Users/isaaclee/Documents/GitHub/ember-container-query/docs-app/my-workspace/ember-container-query/dist/components/container-query.css
[docs-app] my-workspace/docs-app
[docs-app] global
[docs-app] /Users/isaaclee/Documents/GitHub/ember-container-query/docs-app/my-workspace/docs-app/controllers/products.css
[docs-app] my-workspace/docs-app
[docs-app] global
```

To avoid depending too much on [implementation details in Embroider](https://github.com/embroider-build/embroider/blob/v1.8.3/packages/compat/src/default-pipeline.ts#L31-L38), which is currently in flux, can we continue to use `include` instead?

I checked that the latter does solve the issue from using `SAVE_WORKSPACE`. But, of course, there is the tradeoff of `include()` being a weaker assertion than `startsWith()`, i.e. in some other Ember projects, some paths are incorrectly marked as `local` instead of `global`.

```sh
# With SAVE_WORKSPACE
[docs-app] /Users/isaaclee/Documents/GitHub/ember-container-query/docs-app/my-workspace/docs-app/components/widgets/widget-3/tour-schedule.css
[docs-app] my-workspace/docs-app
[docs-app] local
[docs-app] /Users/isaaclee/Documents/GitHub/ember-container-query/docs-app/my-workspace/ember-container-query/dist/components/container-query.css
[docs-app] my-workspace/docs-app
[docs-app] global
[docs-app] /Users/isaaclee/Documents/GitHub/ember-container-query/docs-app/my-workspace/docs-app/components/navigation-menu.css
[docs-app] my-workspace/docs-app
[docs-app] local

# Without SAVE_WORKSPACE
[docs-app] /private/var/folders/bg/0wpjv_sn7891sjq4ch0_3f0w0000gr/T/embroider/280b31/docs-app/controllers/dashboard.css
[docs-app] /private/var/folders/bg/0wpjv_sn7891sjq4ch0_3f0w0000gr/T/embroider/280b31/docs-app
[docs-app] local
[docs-app] /private/var/folders/bg/0wpjv_sn7891sjq4ch0_3f0w0000gr/T/embroider/280b31/docs-app/controllers/products.css
[docs-app] /private/var/folders/bg/0wpjv_sn7891sjq4ch0_3f0w0000gr/T/embroider/280b31/docs-app
[docs-app] local
[docs-app] /private/var/folders/bg/0wpjv_sn7891sjq4ch0_3f0w0000gr/T/embroider/280b31/ember-container-query/dist/components/container-query.css
[docs-app] /private/var/folders/bg/0wpjv_sn7891sjq4ch0_3f0w0000gr/T/embroider/280b31/docs-app
[docs-app] global
```

Maybe such false positives are better dealt on a case-by-case basis (let the end-developer of that project include additional checks).